app/server/lib/Client.ts

```diff
@@ -531,7 +531,18 @@ export class Client {
    * indicate success or failure.
    */
   private async _onMessage(message: string): Promise<void> {
-    const request = JSON.parse(message);
+    let request;
+    try {
+      request = JSON.parse(message);
+    } catch (err) {
+      this._log.warn(null, "failed to parse message: %s", err.toString());
+      if (!this._destroyed) {
+        this._sendToWebsocket(JSON.stringify({error: err.message}))
```

Comment:
I bypass the Memory pool, if that's OK. Mostly because I don't have any `reqId` to pass.

As an alternative, we may just log and not even send a WS response.